src/get_patches/crop_tiles_to_patches.py

```diff
@@ -0,0 +1,171 @@
+                                    download_dir: Path) -> None:
+
+    """
+
+    I was exploring alternative methods to data downloading, this is no better than Murrays method and should
+    not be called preferentially
+
+    This function may be used to get snowpatch images, based on the 9 sentinel bands available, using the
+    sentinelsat api for simplicity
+
+    VERY IMPORTANT: this resolution of data is stored in the long term archive, these requests retrieve from
+    the LTA, the first retrieval may take hours. Once one of us has retrieved a patch, it should stay
+    out of the LTA and be much faster for others
+
+    :param patch_geojson_path: path to any geojson, generate these by running generate_prerequisites.py
+    :param username: copernics username (https://scihub.copernicus.eu/dhus/#/home)
+    :param password: copernicus password
+    :param download_dir: path to store images
+    :return: None, saves images to directory
+    """
+
+    # Connect to the API
+    api = SentinelAPI(username, password, 'https://scihub.copernicus.eu/dhus')
+
+    # Load your area of interest from the GeoJSON
+    footprint = geojson_to_wkt(read_geojson(patch_geojson_path))
+
+    # Search for available products
+    # You can change the date and product type based on your needs
+    products = api.query(footprint,
+                         date=('20230301', '20230630'),
+                         platformname='Sentinel-2',
+                         cloudcoverpercentage=(0, 30))
+
+    # Print found products
+    print(f"{len(products)} products found.")
+
+
+    # for dev purposes
+    uuid = next(iter(products))
+
+    max_retries = 5
+    retry_delay = 3600  # retry after 1 hour
+
+
+    for i in range(max_retries):
+        try:
+            product_info = api.download(uuid, directory_path=download_dir)
+            print(f"Downloaded {product_info['title']}.")
+            break
+        except LTATriggered:
+            print(f"Product is still in LTA. Retry {i + 1}/{max_retries}. Waiting for {retry_delay} seconds.")
+            time.sleep(retry_delay)
+
+
+def clip_sentinel_tile_to_region(tile_path: Path,
+                                 patch_geojson_path: Path) -> None:
+    """
+    Sentinel images come in tiles of 100mk^2, they must be clipped to snowpatches using the geojsons
+    the team has generated.
+
+    VERY IMPORTANT:
+    Here's a breakdown of the Sentinel-2 MSI bands and their respective resolutions:
+
+    10m Resolution Bands:
+
+    Band 2 (Blue)
+    Band 3 (Green)
+    Band 4 (Red)
+    Band 8 (NIR)
+    20m Resolution Bands:
+
+    Band 5 (Red Edge 1)
+    Band 6 (Red Edge 2)
+    Band 7 (Red Edge 3)
+    Band 8A (Red Edge 4)
+    Band 11 (SWIR 1)
+    Band 12 (SWIR 2)
+    60m Resolution Bands:
+
+    Band 1 (Coastal/Aerosol)
+    Band 9 (Water Vapor)
+    Band 10 (Cirrus)
+
+    Therefore, to load in the 10m res bands be sure to load ".....1_B02.jp2", ".....1_B03.jp2" etc
```

Comment:
I have one question about this. When I download the tiles, each band is saved as ..._B03_10m.jp2, so the resolution is also in the filename. I think this could mean that the code on line 112 below doesn't always extract the band correctly?

Reply:
hey Murray, let me fix this now